core/src/services/lakefs/core.rs

```diff
@@ -0,0 +1,166 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+use std::fmt::Debug;
+
+use http::header;
+use http::Request;
+use http::Response;
+use serde::Deserialize;
+
+use crate::raw::*;
+use crate::*;
+
+pub struct LakefsCore {
+    pub endpoint: String,
+    pub repository_id: String,
+    pub branch: String,
+    pub root: String,
+    pub username: String,
+    pub password: String,
+    pub client: HttpClient,
+}
+
+impl Debug for LakefsCore {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("LakefsCore")
+            .field("endpoint", &self.endpoint)
+            .field("username", &self.username)
+            .field("password", &self.password)
+            .field("root", &self.root)
+            .field("repository_id", &self.repository_id)
+            .field("branch", &self.branch)
+            .finish_non_exhaustive()
+    }
+}
+
+impl LakefsCore {
+    pub async fn get_file_meta(&self, path: &str) -> Result<Response<Buffer>> {
```

Comment:
Please try best to align with the upstream API name. For example, this API should call `GetObjectMetadata`:

![image](https://github.com/user-attachments/assets/913c5527-d3a7-4120-b1d8-1ffe7d6c4fb4)